### PR TITLE
Publish to Test PyPI from Travis CI jobs

### LINKED ISSUE #110: Implement auto-publishing new dev dists to Test PyPI on each push to master
Cutting-edge example:
- https://github.com/sanitizers/octomachinery/blob/f7f023d/.travis.yml#L53-L65
- https://github.com/sanitizers/octomachinery/blob/f7f023d/setup.cfg#L9
- https://github.com/sanitizers/octomachinery/blob/f7f023d/octomachinery/utils/versiontools.py#L9-L38

### PR DESCRIPTION
**Pull request by MaxTyulin**
## What do these changes do?
Perform auto-publishing new dev dists to Test PyPI.

## Related issue number
Resolves #110 

## Checklist

- [ ] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` (e.g. `588.bugfix`)
  * if you don't have an `issue_id` change it to the pr id after creating the PR
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: `Fix issue with non-ascii contents in doctest text files.`

### CODE REVIEW

**Comment by MaxTyulin** [comment]
BTW, you need to specify user/password for test pypi account.

File: .travis.yml [hunk 1]
```diff
@@ -27,8 +27,10 @@ cache: pip
 
 jobs:
   include:
-    - stage: Upload a new version of python package to PYPI
+    - &deploy_job
+      stage: Upload a new version of python package to PYPI
       name: Publishing current Git tagged version of dist to PyPI
+      env: PYPI_UPLOAD=true
```
Comment (webknjaz) [inline .travis.yml]: This is only necessary for Test PyPI.

Comment (MaxTyulin) [reply .travis.yml]: Have moved it lower.

File: .travis.yml [hunk 2]
```diff
@@ -48,3 +49,19 @@ jobs:
             ktCdEw9PQO9orA7BmhCn71yOIPK3seomRkJi8+ea54lQsT8cCe+sKZgaXRTBwkcKnKqz11gQl7P11qPNLIWr4cNeVrF6RsDix5YFDxpfqy+x6IPNvpArhkB8/2u2SpFbjzTOh0jT2d4ucSUDV+axjSRPymHK+8lYEMKXucNrzsao8qrExpCMkVJA5rNKeLC537ZN53tVbhwYNih+LsGPXGTvoQoMSQPRk83NfselIper1bPwioUYzIaJCygrOKQKqs/nKOlygMFmPyg4uFoDBh3idG66r8Q9Zmt7DrCAaCxJyGFmjNYHkMzLLh8kl2VuXk+lREbmPDIfNEGzoAM4mX6Jd+LGNvbOOqntRYpW+K4A01qe+YnI142uA7tfxWl1qOqknb0bT6EXKq3iCB4ZXxLdukCcaBAUJP0ctj5wyVpcJ7TAst4SDKLYuZ06I+kx2n51HO5DsfuUQBvkKuMzAHXv0jxXcv8ez2ZS5kBpgQuemNm1NOsrWhtyWzvhlcD2+VMNaYNII63GvAUiwgO8QCf/JNaW12Fdp/nceVUhJWlsmfvQQ1erKqugtecrowh9XWTO5kx7CMrpZJCDdWXMoUMC7aRmJuYaSiz1xCHa44v6xFlMv7bv2UzelRbnpgolCCTzCGXnC6ozct58b/dtGiHKbPJ+cJSOjH513kxRSaM=
         on:
           all_branches: true
+    - <<: *deploy_job
+      name: >-
+        Publishing current (unstable) Git revision of dist to Test PyPI
+        on every commit
+      env: PYPI_UPLOAD=true
```
Comment (webknjaz) [inline .travis.yml]: Please use a mapping instead of a string value.

File: .travis.yml [hunk 3]
```diff
@@ -48,3 +49,19 @@ jobs:
...
+      if: >-  # Always run, except if PR or cron
+        repo == "aio-libs/create-aio-app" AND
+        branch == "master" AND
+        type == "push"
+      deploy:
+        <<: *deploy_step
+        server: https://test.pypi.org/legacy/
+        user: <test-pypi-user>
```
Comment (webknjaz) [inline .travis.yml]: Drop this line because the username would be the same.

Comment (webknjaz) [reply .travis.yml]: Or better change it to `__token__`.
```suggestion
        user: __token__
```

File: setup.cfg [hunk 4]
```diff
@@ -4,8 +4,6 @@ version = attr: create_aio_app.utils.versiontools.get_self_version
 description = The tool that helps quickly create a basis for aiohttp application
 long_description = file: README.md
 long_description_content_type = text/markdown
-platforms =
```
Comment (webknjaz) [inline setup.cfg]: This change is unrelated to publishing to PyPI. Revert it.

File: setup.cfg [hunk 5]
```diff
@@ -4,7 +4,7 @@ version = attr: create_aio_app.utils.versiontools.get_self_version
 description = The tool that helps quickly create a basis for aiohttp application
 long_description = file: README.md
 long_description_content_type = text/markdown
-platforms = 
+platforms =
```
Comment (webknjaz) [inline setup.cfg]: This shouldn't show up in the PR, adjust that space.

File: .travis.yml [hunk 6]
```diff
@@ -48,3 +49,20 @@ jobs:
...
+        repo == "aio-libs/create-aio-app" AND
+        branch == "master" AND
+        type == "push"
+      deploy:
+        <<: *deploy_step
+        server: https://test.pypi.org/legacy/
+        user: __token__
+        password:
+          secure: >-
+            <encrypted-test-pypi-user-password>
```
Comment (webknjaz) [inline .travis.yml]: I'll generate and encrypt the token. And then, I'll post it here.

File: setup.cfg (hunk 5 above)
Comment (MaxTyulin) [reply setup.cfg]: Done.

File: .travis.yml (hunk 6 above)
Comment (webknjaz) [reply .travis.yml]: Here you go:
```suggestion
            rfpMPtYl0ipSV2ckqiiLPwfx28esAf4zybAGk2imd5zRVI2TudaHPna0mcLQCJ0Rn5rdl3eBdRfTFoFl+V60+5d98SpwE8PCxlY5hAwXd/CirmpjjKj3N8pvIUrs1KHll0gxaMCXEPmEatiWlkF8av4ZUX/ycytQZE3men+Nw4wVMbruKcMoRkTMm1AJ3u1g0lOd4Dr8Gv2mc4vlKvk6a4v8abKsi8mfHOxUDoq5dQdIl+Ghht/eCSb3lGop5o8D1E+7jZv4wj5k2JVpU0ulb7sOKMdzKls0XKINekOroIg3sgIpdzuEyKfW64tIv8PZGYaZrQgb4jIKmwtklbfC6fCY/LkchsdskPO/1QsuLiOIp+JsmAuDfZNZvZu8xSIvmMhmkHPY9ioxZydHWtwvcrfoAXHvCX90v7xlzqqSruAbYfyoZpsDKnJ+FcRVr6x1h+BQHT0kg9k39bGWOyzdJhDt4WaifiM81G5t7zeQTVKiTGWaNRuaE0xFe1kyY4vMYj63snQuAuw8i2zbJ4AubQSHyf4X+9Emgna+/wnNIbVZAaXsf8PJyXP+9JOusGvIg0I3q/cW0v21eXUj4+AIDTXOhdovndurbkHw5otDpRfD4vt0Hdh+YaYmOiOEOh4yBFlnMuQlV5QJwwb50QJDgipExrkYHq7cfNifoPLHWhw=
```
(just click "Accept" under this suggestion)

File: .travis.yml [hunk 7]
```diff
@@ -48,3 +49,20 @@ jobs:
             ktCdEw9PQO9orA7BmhCn71yOIPK3seomRkJi8+ea54lQsT8cCe+sKZgaXRTBwkcKnKqz11gQl7P11qPNLIWr4cNeVrF6RsDix5YFDxpfqy+x6IPNvpArhkB8/2u2SpFbjzTOh0jT2d4ucSUDV+axjSRPymHK+8lYEMKXucNrzsao8qrExpCMkVJA5rNKeLC537ZN53tVbhwYNih+LsGPXGTvoQoMSQPRk83NfselIper1bPwioUYzIaJCygrOKQKqs/nKOlygMFmPyg4uFoDBh3idG66r8Q9Zmt7DrCAaCxJyGFmjNYHkMzLLh8kl2VuXk+lREbmPDIfNEGzoAM4mX6Jd+LGNvbOOqntRYpW+K4A01qe+YnI142uA7tfxWl1qOqknb0bT6EXKq3iCB4ZXxLdukCcaBAUJP0ctj5wyVpcJ7TAst4SDKLYuZ06I+kx2n51HO5DsfuUQBvkKuMzAHXv0jxXcv8ez2ZS5kBpgQuemNm1NOsrWhtyWzvhlcD2+VMNaYNII63GvAUiwgO8QCf/JNaW12Fdp/nceVUhJWlsmfvQQ1erKqugtecrowh9XWTO5kx7CMrpZJCDdWXMoUMC7aRmJuYaSiz1xCHa44v6xFlMv7bv2UzelRbnpgolCCTzCGXnC6ozct58b/dtGiHKbPJ+cJSOjH513kxRSaM=
         on:
           all_branches: true
+    - <<: *deploy_job
+      name: >-
+        Publishing current (unstable) Git revision of dist to Test PyPI
+        on every commit
```
Comment (webknjaz) [inline .travis.yml]: Actually, it's on every push...

Comment (MaxTyulin) [reply .travis.yml]: Updated.

**Comment by webknjaz** [comment]
Great, thanks! We also need to use an API token for normal PyPI: https://github.com/aio-libs/create-aio-app/issues/157. (via a separate PR)

**Comment by webknjaz** [comment]
Now, let's check out how it works: https://travis-ci.com/aio-libs/create-aio-app/jobs/249825060
